content/post/kubestash-v2024.6.4/index.md

```diff
@@ -0,0 +1,258 @@
+
+Here is an example of a `BackupConfiguration` for backing up both the manifests and data of a `StatefulSet` in the same session:
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: BackupConfiguration
+metadata:
+  name: sample-backup
+  namespace: demo
+spec:
+  target:
+    apiGroup: apps
+    kind: StatefulSet
+    name: sample-sts
+    namespace: demo
+  backends:
+  - name: s3-storage
+    storageRef:
+      namespace: demo
+      name: s3-storage
+    retentionPolicy:
+      name: demo-retention
+      namespace: demo
+  sessions:
+  - name: workload-backup
+    sessionHistoryLimit: 3
+    scheduler:
+      schedule: "*/5 * * * *"
+      jobTemplate:
+        backoffLimit: 1
+    repositories:
+    - name: demo-storage
+      backend: s3-storage
+      directory: /sts
+      encryptionSecret:
+        name: encry-secret
+        namespace: demo
+    addon:
+      name: workload-addon
+      tasks:
+      - name: manifest-backup
+      - name: logical-backup
+        params:
+          paths: /source/data
+          exclude: /source/data/tmp
+    retryConfig:
+      maxRetry: 2
+      delay: 1m
+```
+
+Here's an example of a `RestoreSession` that restores `StatefulSet` manifests and then restores its data:
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: RestoreSession
+metadata:
+  name: sample-restore
+  namespace: demo
+spec:
+  manifestOptions:
+    restoreNamespace: dev
+  dataSource:
+    repository: demo-storage
+    snapshot: latest
+    encryptionSecret:
+      name: encry-secret
+      namespace: demo
+  addon:
+    name: workload-addon
+    tasks:
+    - name: manifest-restore
+    - name: logical-backup-restore
+```
+
+In this case, a helper `RestoreSession` will be created in the same namespace of the applied `RestoreSession` to restore the data of the workload.
+
+Here is an example of a `BackupConfiguration` for backing up both the manifests and data (dump) of a `MySQL` database in the same session:
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: BackupConfiguration
+metadata:
+  name: mysql-backup
+  namespace: demo
+spec:
+  target:
+    apiGroup: kubedb.com
+    kind: MySQL
+    namespace: demo
+    name: mysql-backup
+  sessions:
+  - name: frequent-backup
+    sessionHistoryLimit: 3
+    scheduler:
+      schedule: "*/5 * * * *"
+      successfulJobsHistoryLimit: 1
+      failedJobsHistoryLimit: 1
+      jobTemplate:
+        backoffLimit: 1
+    repositories:
+    - name: mysql-storage
+      directory: /mysql
+      encryptionSecret:
+        name: encry-secret
+        namespace: demo
+    addon:
+      name: mysql-addon
+      tasks:
+      - name: logical-backup
+      - name: manifest-backup
+    retryConfig:
+      maxRetry: 2
+      delay: 1m
+```
+
+Here's an example of a `RestoreSession` that restores `MySQL` manifests and then restores its data:
```

Comment:
Here is an example of a RestoreSession that first redeploys the MySQL database from the backed-up manifests and then restores the data into it: